### PR TITLE
utoipa: Replace `assert-json-diff` with snapshot testing via `insta`

### PR DESCRIPTION
**Pull request by Turbo87**
- same as #1253, but for the `utoipa` crate this time

### CODE REVIEW

File: utoipa/src/openapi.rs [hunk 1]
```diff
@@ -795,61 +780,10 @@ mod tests {
...
-                      }
-                    }
-                  }
-                }
-            )
-        )
+
+        assert_json_snapshot!(api_1, {
+            ".paths" => insta::sorted_redaction()
+        });
```
Comment (juhaku) [inline utoipa/src/openapi.rs:786]: Does this keep the `paths` sorted always in same order?

**Review by juhaku — APPROVED** [review]
Super, looks good :+1:

File: utoipa/src/openapi.rs (hunk 1 above)
Comment (Turbo87) [reply utoipa/src/openapi.rs:786]: yep. the `merge()` call caused them to be a bit random for some reason, but this fixed it.

Comment (Turbo87) [reply utoipa/src/openapi.rs:786]: https://insta.rs/docs/redactions/ has more info on this :)

**Pull request merged**